website/www/site/layouts/partials/section-menu/en/documentation.html

```diff
@@ -1,367 +1,368 @@
+         <li><a href="/documentation/programming-guide/#user-initiated-checkpoint">User-initiated checkpoint</a></li>
+         <li><a href="/documentation/programming-guide/#runner-initiated-split">Runner initiated split</a></li>
+         <li><a href="/documentation/programming-guide/#watermark-estimation">Watermark estimation</a></li>
+         <li><a href="/documentation/programming-guide/#truncating-during-drain">Truncating during drain</a></li>
+         <li><a href="/documentation/programming-guide/#bundle-finalization">Bundle finalization</a></li>
+       </ul>
+     </li>
+ 
+     <li class="section-nav-item--collapsible">
+       <span class="section-nav-list-title">Multi-language Pipelines</span>
+ 
+       <ul class="section-nav-list">
+         <li><a href="/documentation/programming-guide/#create-x-lang-transforms">Creating cross-language transforms</a></li>
+         <li><a href="/documentation/programming-guide/#use-x-lang-transforms">Using cross-language transforms</a></li>
+         <li><a href="/documentation/programming-guide/#x-lang-transform-runner-support">Runner Support</a></li>
+       </ul>
+     </li>
+     <li><a href="/documentation/programming-guide/#batched-dofns">Batched DoFns</a></li>
+   </ul>
+ </li>
+ <li class="section-nav-item--collapsible">
+   <span class="section-nav-list-title">Pipeline development lifecycle</span>
+   <ul class="section-nav-list">
+     <li><a href="/documentation/pipelines/design-your-pipeline/">Design Your Pipeline</a></li>
+     <li><a href="/documentation/pipelines/create-your-pipeline/">Create Your Pipeline</a></li>
+     <li><a href="/documentation/pipelines/test-your-pipeline/">Test Your Pipeline</a></li>
+   </ul>
+ </li>
+ <li class="section-nav-item--collapsible">
+   <span class="section-nav-list-title">Common pipeline patterns</span>
+ 
+   <ul class="section-nav-list">
+     <li><a href="/documentation/patterns/overview/">Overview</a></li>
+     <li><a href="/documentation/patterns/file-processing/">File processing</a></li>
+     <li><a href="/documentation/patterns/side-inputs/">Side inputs</a></li>
+     <li><a href="/documentation/patterns/pipeline-options/">Pipeline options</a></li>
+     <li><a href="/documentation/patterns/custom-io/">Custom I/O</a></li>
+     <li><a href="/documentation/patterns/custom-windows/">Custom windows</a></li>
+     <li><a href="/documentation/patterns/bigqueryio/">BigQueryIO</a></li>
+     <li><a href="/documentation/patterns/ai-platform/">AI Platform</a></li>
+     <li><a href="/documentation/patterns/schema/">Schema</a></li>
+     <li><a href="/documentation/patterns/bqml/">BigQuery ML</a></li>
+     <li><a href="/documentation/patterns/grouping-elements-for-efficient-external-service-calls/">Grouping elements for efficient external service calls</a></li>
+   </ul>
+ </li>
+ <li class="section-nav-item--collapsible">
+   <span class="section-nav-list-title">AI/ML pipelines</span>
```

Comment:
Note that this section of this page will change in https://github.com/apache/beam/pull/25700/files. If that PR is merged first, add the new topic to whichever category in the left nav corresponds to the table in the Overview page.